roles/openshift_docker/tasks/main.yml

```diff
@@ -0,0 +1,28 @@
+---
+# It's important that we don't explicitly pull this image here.  Otherwise we
+# could result in upgrading a preinstalled environment.  We'll have to set
+# openshift_image_tag correctly for upgrades.
+- name: Set version when containerized
+  command: >
+    docker run --rm {{ openshift.common.cli_image }}:latest version
+  register: cli_image_version
+  when: openshift.common.is_containerized | bool and openshift_image_tag is not defined
```

Comment:
I didn't want to do anything with openshift_image_tag or image tags in openshift facts simply because the only way it can be computed is post docker install.  And I thought it made since to have the place where it was computed close to the place that handles the overriding.  I ripped all the image_tag stuff out of openshift facts.  I think soon we need to fix how openshift_version is used all over the place to be a canonical format and not have to use image_tag_to_rpm_version all over the place.  We have a bit of a mess right now.